lib/cog/models/relay.ex

```diff
@@ -24,11 +24,24 @@ defmodule Cog.Models.Relay do
     model
     |> Repo.preload(:groups)
     |> cast(params, @required_fields, @optional_fields)
+    |> allow_user_defined_id_on_insert(params)
     |> validate_presence_on_insert(:token)
     |> encode_token
     |> unique_constraint(:name)
   end
 
+  def allow_user_defined_id_on_insert(changeset, params) do
+    case {get_field(changeset, :id), Map.get(params, "id")} do
```

Comment:
If we add `:id` to `@optional_fields` can we use `get_change` here instead of passing in params? Sounded like you may have already tried that yesterday, but just wanted to make sure.